### DISCUSSION TITLE
Lockdown use of SF Inspector reloaded to only System Admins?

### DISCUSSION THREAD

**Discussion by megmoo** [tprouvot/Salesforce-Inspector-reloaded#433] [Ideas]
I love this app and am so grateful for all the work that has been put into it. Thank you so much. 

Would it be possible to add some way to allow System Admins to configure our Salesforce org so that only those with the System Administrator profile can use any of the data editing/deleting features? The goal is to provide read-only functionality only for non Sys Admins. Ideally, you could either restrict non-admin users all together or to limit them to read-only features.  

Even if it's something as hacky as implementing this access control when the Org's Division is set to various strings like "SF Inspector Reloaded Admin Access Only" or "SF Inspector Reloaded Read-Only For Non-Admins."

**Comment by tprouvot** [comment]
Hi @megmoo,
Today this is not possible, however you [can restrict the use of the extension to specific users through permission set or profile](https://tprouvot.github.io/Salesforce-Inspector-reloaded/how-to/#use-sf-inspector-with-a-connected-app) via the usage of a connected App.

The difference is, those without the rights won't be able to use the extension at all.
What are the reasons/usage that lead you to allow a read only access to the extension (instead of not access at all) ?

**Reply by megmoo** [reply]
The main requirement is restricting to only certain users, so this is fantastic news! Thank you!

The only reason to restrict to read-only is that we have some "super users" who would benefit from the querying functionality, but we wouldn't want them to have the amazing powers of mass updates and deletes at their fingertips.
